runner.bat.template

```diff
@@ -0,0 +1,43 @@
+@echo off
+setlocal EnableDelayedExpansion
+
+set BUILDIFIER_SHORT_PATH=@@BUILDIFIER_SHORT_PATH@@
+set ARGS=@@ARGS@@
+set WORKSPACE=@@WORKSPACE@@
+
+REM Get the absolute path to the buildifier executable
+for %%I in ("%BUILDIFIER_SHORT_PATH%") do set "buildifier_short_path=%%~fI"
+
+REM Use TEST_WORKSPACE to determine if the script is being ran under a test
+if defined TEST_WORKSPACE if not defined BUILD_WORKSPACE_DIRECTORY (
+    set "FIND_FILE_TYPE=l"
+    
+    REM If WORKSPACE was provided, then the script is being run under a test in no_sandbox mode
+    REM cd to the directory containing the WORKSPACE file
+    if defined WORKSPACE (
+        set "FIND_FILE_TYPE=f"
+        for %%I in ("%WORKSPACE%") do set "WORKSPACE_PATH=%%~dpI"
+        cd /d "!WORKSPACE_PATH!"
+        if not exist "!WORKSPACE_PATH!" (
+            echo Unable to change to workspace (WORKSPACE_PATH: !WORKSPACE_PATH!)
+        )
+    )
+) else (
+    REM Change into the workspace directory if this is _not_ a test
+    cd /d "%BUILD_WORKSPACE_DIRECTORY%" || (
+        echo Unable to change to workspace (BUILD_WORKSPACE_DIRECTORY: %BUILD_WORKSPACE_DIRECTORY%)
+        exit /b 1
+    )
+)
+
+REM Run buildifier on all starlark files
+for /r %%F in (*.bzl *.sky *.bazel BUILD *.BUILD BUILD.*.oss WORKSPACE WORKSPACE.bzlmod WORKSPACE.oss WORKSPACE.*.oss) do (
+    if "%FIND_FILE_TYPE%" == "l" (
+        set "fileType=%%~aF"
+        if "!fileType:~0,1!" == "l" (
+            "!buildifier_short_path!" %ARGS% "%%~fF"
```

Comment:
i don't know bat at all but does this invoke buildifier once per discovered file? if so is that going to be a performance problem? the bash version collects all the files and passes them to only 1 exec of buildifier

Reply:
Fixed, thanks for catching that!